community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -706,6 +742,12 @@ public IndexMap apply( IndexMap indexMap ) throws IOException
             for ( IndexRule rule : rules )
             {
                 long ruleId = rule.getId();
+                if ( state == State.NOT_STARTED )
+                {
+                    // In case of recovery remove any previously recorded INDEX DROP for this particular index rule id,
+                    // in some scenario where rule ids may be reused.
+                    indexesToDropAfterCompletedRecovery.remove( ruleId );
```

Comment:
Can we add a test for this scenario?

Reply:
Perhaps a unit test, because I'm not entirely sure it's even possible on a higher level... but not 100%. A unit test it is...